sklearn/base.py

```diff
@@ -26,7 +27,7 @@ def _first_and_last_element(arr):
         return arr[0, 0], arr[-1, -1]
 
 
-def clone(estimator, safe=True):
+def clone(estimator, safe=True, deepcopy=None):
```

Comment:
I've been preferring `deepcopy='warn'` and similar lately over None.

I wonder if we can instead use magic to check if the caller is internal to this library, and if so, just replace `deepcopy='warn'` with `deepcopy=False`.

```
import inspect

def is_caller_sklearn():
    caller = inspect.current_frame().f_back.f_back
    return caller is not None and caller.f_globals['__name__'].startswith('sklearn.')
```